Add daily exercises.

eschlange opened the pull request:
* Added specified exercise descriptions and content.
* Added cache for exercise data.
* Added exercise service.
* Added view for exercises.
* Added controller for exercises.
* Added specs for exercise cache and controller methods.
* Updated home view to use new exercise data.
* Updated routes and configuration files for new logic.

[#95522472]

michaelwehrley commented on src/js/services/Exercise.js:
```diff
@@ -0,0 +1,63 @@
+    this.updateExercise = function (exercise) {
+      debugger
+      var exerciseData = exerciseCache.fetchAllRaw();
+      for (var i = 0; exerciseData && i < exerciseData.length; i++) {
```
You have debugger on line 42, and could you use https://docs.angularjs.org/api/ng/function/angular.forEach here?  If you chose the incrementing over the each method, why?  Thanks.
```javascript
angular.forEach(exerciseData, function(value, key) {
```

eschlange replied:
I could, but then I'd have to write a wrapping conditional to check for an undefined exerciseData, angular expects the object passed in to not be undefined. The for loop takes care of both.

ericcf replied:
Maybe we should update `fetchAllRaw` to return an empty array instead. You could also use

```javascript
angular.forEach((exerciseData || []), function(value, key) {
```

michaelwehrley commented on test/unit/controllers/ExercisesControllerSpec.js:
```diff
@@ -0,0 +1,35 @@
+      controller.todaysExercise.answers.push("some answer");
+      controller.todaysExercise.answers.push("another answer");
+      controller.saveExercise();
+      expect(controller.todaysExercise.answers.length).to.equal(2);
```
Because answers is an array, by pushing all this into the array, if you removed line 23 would it still pass?

michaelwehrley commented:
:eyes: great to me!

ericcf commented:
@michaelwehrley do you think unit tests are needed for the `Excercise` service?

eschlange commented:
@ericcf @michaelwehrley I forgot to include my exercises for the 'Exercise' service. I'll commit and push those shortly.

eschlange commented:
@michaelwehrley Please take a look at my changes. I made the updates that you recommended, added exercise service test cases, and refactored my existing ones.

michaelwehrley commented:
:+1:
